src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/PurposesApiServiceImpl.scala

```diff
@@ -394,9 +387,9 @@ final case class PurposesApiServiceImpl(
       purposeUUID       <- purposeId.toFutureUUID
       purpose           <- purposeProcessService.getPurpose(purposeUUID)
       eService          <- catalogProcessService.getEServiceById(purpose.eserviceId)
-      agreement         <- getLatestAgreement(purpose.consumerId, eService, ec, agreementProcessService).flatMap(
-        _.toFuture(new Exception())
-      )
+      agreement         <- agreementProcessService
+        .getLatestAgreement(purpose.consumerId, eService)
+        .flatMap(_.toFuture(AgreementNotFound(purpose.consumerId)))
```

Comment:
Does this exception need to be handled in the handleΕrrorXX function?

Reply:
good point, but probably not. An agreement should always exist for a purpose (no filters are applied in the retrieve). 
If no agreement is found, it's worth to return a 500 and get notified